substrate/frame/support/procedural/src/pallet/expand/genesis_config.rs

```diff
@@ -104,44 +56,6 @@ pub fn expand_genesis_config(def: &mut Def) -> proc_macro2::TokenStream {
 			attrs.push(syn::parse_quote!( #[serde(bound(deserialize = ""))] ));
 			attrs.push(syn::parse_quote!( #[serde(crate = #serde_crate)] ));
 		},
-		_ => unreachable!("Checked by genesis_config parser"),
-	}
-
-	quote::quote! {
-		#[doc(hidden)]
-		pub mod __substrate_genesis_config_check {
-			#[macro_export]
-			#[doc(hidden)]
-			macro_rules! #def_macro_ident {
-				($pallet_name:ident) => {};
-			}
-
-			#[cfg(not(feature = "std"))]
-			#[macro_export]
-			#[doc(hidden)]
-			macro_rules! #std_macro_ident {
-				($pallet_name:ident, $pallet_path:expr) => {
-					compile_error!(concat!(
-						"`",
-						stringify!($pallet_name),
-						"` does not have the std feature enabled, this will cause the `",
-						$pallet_path,
-						"::GenesisConfig` type to not implement serde traits."
-					));
-				};
-			}
-
-			#[cfg(feature = "std")]
-			#[macro_export]
-			#[doc(hidden)]
-			macro_rules! #std_macro_ident {
-				($pallet_name:ident, $pallet_path:expr) => {};
-			}
-
-			#[doc(hidden)]
-			pub use #def_macro_ident as is_genesis_config_defined;
-			#[doc(hidden)]
-			pub use #std_macro_ident as is_std_enabled_for_genesis;
-		}
```

Comment:
factorized out into tt_default_parts.

Note there is a special check for `std` feature. it is also generated in the dummy parts logic.